save-sandbox/src/main/kotlin/com/saveourtool/save/sandbox/security/ConvertingAuthenticationManager.kt

```diff
@@ -0,0 +1,66 @@
+package com.saveourtool.save.sandbox.security
+
+import com.saveourtool.save.sandbox.service.SandboxUserDetailsService
+import com.saveourtool.save.sandbox.utils.extractUserNameAndIdentitySource
+import com.saveourtool.save.utils.AuthenticationDetails
+import com.saveourtool.save.utils.IdentitySourceAwareUserDetails
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.security.authentication.ReactiveAuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+/**
+ * Implementation of `ReactiveAuthenticationManager` that doesn't check user credentials
+ * and simply authenticates every user with username. Should be used in secure environment,
+ * where user identity is already guaranteed.
+ */
+@Component
+class ConvertingAuthenticationManager : ReactiveAuthenticationManager {
+    @Autowired
+    private lateinit var sandboxUserDetailsService: SandboxUserDetailsService
```

Comment:
can we use dependency injection in constructor?

Reply:
Moved